### ISSUE TITLE
Enable ClusterTopology by default, and remove feature gate

### ISSUE THREAD

**Issue by vincepri** [kubernetes-sigs/cluster-api#8542]
With the 1.4 release, we've brought ClusterTopology and its related objects a few steps forward. While these flags disable the use of the CRDs, the actual APIs still have v1beta1 guarantees. 

This issue tracks the removal of the feature gate in 1.5, and potentially starting to enable the feature by default in a patch version of 1.4.

**Comment by nikzayn** [comment]
Hey @vincepri I would like to take this issue. Actually, can you brief me more about this like as I know from the subject heading it says that enable cluster topology by default, I just wanted to know at which place!

**Comment by fabriziopandini** [comment]
I'm generally +1, but I'm wondering if we should get support for machine pools before graduating.

**Comment by vincepri** [comment]
Given that the APIs are already beta, why wait to remove the feature gate?

**Comment by sbueringer** [comment]
Given that the APIs always were beta, why did we introduce the feature gate?

**Comment by sbueringer** [comment]
Or to ask differently. What does a feature gate / experimental even mean? :)

Why do we have CRDs like ClusterResourceSet, MachinePool and ClusterClass behind a feature gate and call (some of) them experimental?

**Comment by vincepri** [comment]
I see it fit on a case by case basis. For topology specifically: there are fields on a Cluster that were disabled unless opted in, things seem to be a bit more stable code-wise that we can enable them by default without the need of a feature gate.

For other experiments/gates, I'd defer to the other feature maintainers.

**Comment by sbueringer** [comment]
Regarding stability of Cluster.spec.topology. We're just about ready to add new fields for MachinePools: https://github.com/kubernetes-sigs/cluster-api/pull/8820

Currently discussing if we should cherry-pick into v1.5.x once the implementation is there.

I don't know how stable it will be directly after the initial implementation.

**Comment by vincepri** [comment]
That's fine, the fields in the APIs can't have breaking changes anyway? Do we see any reason to keep the gate?

**Comment by sbueringer** [comment]
To be honest I have no idea what our exact criteria are. But if we drop the gate now the cherry-pick into v1.5 that Cecile was suggesting won't be possible anymore as the feature won't be experimental anymore
(cc @CecileRobertMichon)

**Comment by CecileRobertMichon** [comment]
What's the rush?

My vote is to get MachinePool support in before removing the feature gate (we can remove the feature gate early in the v1.6 release cycle once MachinePool support is merged)

**Comment by vincepri** [comment]
sgtm, let's aim for v1.6 — main reason is that I'd like to see increased overall adoption/feedback on clusterclass and managed topologies. It's completely fine to wait for MachinePool, and probably better; this issue is just to track the removal, given that the gate has been around for a while.

/milestone v1.6

**Comment by furkatgofurov7** [comment]
@vincepri @sbueringer @CecileRobertMichon checking in here, since it is in v1.6 milestone: #8820 is merged now, should this be tracked now to remove the feature gate?

**Comment by sbueringer** [comment]
I think we continue to aim to graduate with v1.6. #8820 only merged the API though, there are other PRs that we need for the implementation.

**Comment by neolit123** [comment]
+1 to have a discussion, consider potential major blockers and try to get this out of experimental. in the next release or the one after that. happy to provide feedback about prod usage.

**Comment by dtzar** [comment]
There are currently [46 issues which have "Clusterclass" listed ](https://github.com/kubernetes-sigs/cluster-api/issues?page=2&q=is%3Aissue+is%3Aopen+clusterclass).   Some of these are docs or tangent related.  I suggest someone combs through these and considers if any of these should be deemed required to move out of experimental.  I would suggest not adding new features to Clusterclass as a gate criteria unless there is unanimous support.

**Comment by sbueringer** [comment]
>  I would suggest not adding new features to Clusterclass as a gate criteria unless there is unanimous support.

Sorry, I"m not sure I"m interpreting this correctly. Can you rephrase?

**Comment by dtzar** [comment]
I'm saying the current feature set seems to work well for people today and I wouldn't want to add new features/functionality as a gate to move out of experimental unless there are a few people who agree (e.g. not just one person who thinks we should add this new functionality).  

That said - if one of the current proposed features we think makes obvious sense to incorporate in the near-future AND it has a significant breaking change to what exists, then we should definitely discuss.

**Comment by sbueringer** [comment]
Okay got it. That makes sense!

**Comment by fabriziopandini** [comment]
AFAIK there is no issues/missing features preventing us to move this feature out of experimental, but +1 to make another pass on the open issues

**Comment by sbueringer** [comment]
Renamed the issue so we can use it as a tracking issue for the feature / feature gate going forward
